state/execution.go

```diff
@@ -580,14 +579,9 @@ func validateValidatorUpdates(abciUpdates []abci.ValidatorUpdate,
 		}
 
 		// Check if validator's pubkey matches an ABCI type in the consensus params
-		pk, err := cryptoenc.PubKeyFromProto(valUpdate.PubKey)
-		if err != nil {
-			return err
-		}
-
-		if !types.IsValidPubkeyType(params, pk.Type()) {
+		if !types.IsValidPubkeyType(params, valUpdate.PubKeyType) {
```

Comment:
again, here we're checking only if the type is one of the ones in `ValidatorParams`, but since in `ValidatorUpdate` you have the bytes, you might be able to add some logic to check if the size is also right (matching the type size using the `crypto` library) as a defensive practice since this will be coming from an ABCI response